src/Aspire.Dashboard/Model/ThemeManager.cs

```diff
@@ -71,11 +77,15 @@ public string EffectiveTheme
         set => _effectiveTheme = value;
     }
 
-    public async Task EnsureEffectiveThemeAsync()
+    public async Task EnsureBrowserThemeLoadedAsync()
     {
-        if (_effectiveTheme == null)
+        if (!_hasLoaded)
         {
-            _effectiveTheme = await _effectiveThemeResolver.GetEffectiveThemeAsync(CancellationToken.None).ConfigureAwait(false);
+            var browserThemeSettings = await _themeResolver.GetThemeSettingsAsync(CancellationToken.None).ConfigureAwait(false);
+            _effectiveTheme = browserThemeSettings.EffectiveTheme;
+            SelectedTheme = !string.IsNullOrEmpty(browserThemeSettings.SelectedTheme) ? browserThemeSettings.SelectedTheme : null;
```

Comment:
Should the `SelectedTheme` getter be changed to throw if `!_hasLoaded`, similar to the getter for `EffectiveTheme`?